src/scheduler/MpiWorld.cpp

```diff
@@ -1144,12 +1144,14 @@ void MpiWorld::checkRankOnThisHost(int rank)
     }
 }
 
-void MpiWorld::createWindow(const faabric_win_t* window, uint8_t* windowPtr)
+void MpiWorld::createWindow(const int winRank,
```

Comment:
This also gets called from `faasm/src/wavm/mpi.cpp` as a consequence it can't depend on `faabric_win_t`. The fact we used the `worldMutex` also made it clone a `faasm`-native version.

Luckily, the implementation is simple enough to workaround not passing the actual struct, just what we need.